components/table/demo/table-test.js

```diff
@@ -153,38 +154,40 @@ class TestTable extends RtlMixin(DemoPassthroughMixin(TableWrapper, 'd2l-table-w
 		const desc = e.target.hasAttribute('desc');
 		this._sortDesc = field === this._sortField ? !desc : false; // if sorting on same field then reverse, otherwise sort ascending
 		this._sortField = field;
+		this._handleSortData();
+	}
+
+	_handleSortComplex(e) {
+		this._sortField = e.target?.getAttribute('data-field');
+		this._sortDesc = e.target?.hasAttribute('data-desc');
+		this._handleSortData();
+	}
 
+	_handleSortData() {
 		this._data = this._data.sort((a, b) => {
-			if (this._sortField === 'city' || this._sortField === 'country') {
-				if (this._sortDesc) {
-					if (a[this._sortField] > b[this._sortField]) return -1;
-					if (a[this._sortField] < b[this._sortField]) return 1;
-				} else {
-					if (a[this._sortField] < b[this._sortField]) return -1;
-					if (a[this._sortField] > b[this._sortField]) return 1;
-				}
+			if (this._sortDesc) {
+				if (a.data[this._sortField] > b.data[this._sortField]) return -1;
+				if (a.data[this._sortField] < b.data[this._sortField]) return 1;
 			} else {
-				if (this._sortDesc) {
-					return b.data[this._sortField] - a.data[this._sortField];
-				}
-				return a.data[this._sortField] - b.data[this._sortField];
+				if (a.data[this._sortField] < b.data[this._sortField]) return -1;
+				if (a.data[this._sortField] > b.data[this._sortField]) return 1;
 			}
+			return 0;
 		});
 	}
 
-	_renderDoubleSortButton(item1, item2) {
+	_renderDoubleSortButton(name) {
+		const noSort = this._sortField?.toLowerCase() !== 'city' && this._sortField?.toLowerCase() !== 'country';
 		return html`
 			<th scope="col">
 				<d2l-table-col-sort-button
-					@click="${this._handleSort}"
-					source-type="words"
-					?desc="${this._sortDesc}"
-					?nosort="${this._sortField !== item1.toLowerCase()}">${item1}</d2l-table-col-sort-button>
-				<d2l-table-col-sort-button
-					@click="${this._handleSort}"
-					source-type="words"
 					?desc="${this._sortDesc}"
-					?nosort="${this._sortField !== item2.toLowerCase()}">${item2}</d2l-table-col-sort-button>
+					?nosort="${noSort}">${name}
+					<d2l-menu-item-radio slot="items" text="City, A to Z" data-field="city" @d2l-menu-item-select="${this._handleSortComplex}" value="1"></d2l-menu-item-radio>
+					<d2l-menu-item-radio slot="items" text="City, Z to A" data-field="city" data-desc @d2l-menu-item-select="${this._handleSortComplex}" value="2"></d2l-menu-item-radio>
```

Comment:
This would be my suggested usage:
- `data-field` (follows how the LMS handles this (uses `data-fieldname`)) to get the field to sort on
- `data-desc` to get whether that should trigger adding `desc` or not to the parent `d2l-table-col-sort-button`

This would be decided by the consumer since they have to handle the select event etc. Another option would be to create something like `d2l-table-col-sort-button-item` that extends `MenuItemMixin` and could then have these as properties. Any preference?

Reply:
I have mixed feelings about having consumers place `d2l-menu-item-radio` into `d2l-table-col-sort-button`'s slot. It's effectively surfacing the internal `d2l-menu` usage as part of `d2l-table-col-sort-button`'s public API. I don't expect that we'd change that, but I think I prefer it if the consumer provides all the menu bits (inc `d2l-menu`), or none of it (using `d2l-table-col-sort-button-item` like you mentioned). If we provide a special item component, we'd also want to dispatch a corresponding event, keeping the menu-item even under wraps.

Reply:
That's a good point. I think in that case my personal preference would be to have a special item component that gets used. We'd want the consumer to implement something very specific (I'm especially thinking of the accessibility behaviour here) but we'd be giving the option to do basically anything.